src/tooltip/tooltip.ts

```diff
@@ -249,8 +253,13 @@ export class NgbTooltip implements OnInit, OnDestroy, OnChanges {
 				Promise.resolve().then(() => {
 					// This update is required for correct arrow placement
 					this._positioning.update();
-					this._zoneSubscription = this._ngZone.onStable.subscribe(() => this._positioning.update());
 				});
+				this._afterRenderRef = afterRender(
+					() => {
+						this._positioning.update();
+					},
+					{ phase: AfterRenderPhase.MixedReadWrite, injector: this._injector },
```

Comment:
`MixedReadWrite` is the default value, any reason for specifying it here ? 
Also haven't looked much into it, but its usage isn't recommended as it perf sensitive.

Reply:
I felt like specifying it allowed to make it clear that choosing that value was a conscious choice and not an oversight. I'm not entirely sure what update() does, but I guessed it had to read and write to the DOM, hence this option. 
Given that it's just a single tooltip positioning and that ther's nothing else being read or written to the DOM at the same time, I don't think it matters much, TBH.